### PR TITLE
ADAPT-4026 | @mattanglin | staging site

### PR DESCRIPTION
**Pull request by mattanglin**
# READY FOR REVIEW

# Summary
This PR is infrastructure for having a "staging" or "preview" environment for the SAA Alumni Site. In this PR i've added a new workflow that will sync our `staging` branch with `dev` anytime anything is pushed to `dev`. I've setup a Netlify branch-deploy for our `staging` branch so once `dev` is synced with `staging` Netlify will kick off a `staging` deploy. The build config for `staging` includes a Storyblok preview api-key which will pull content in "Draft" status from our PROD Storyblok space. I've additionally added a custom Netlify plugin that will conditionally add auth headers based on deploy context/branch so that our deployed staging environment will have some basic protection.

- Added custom plugin for netlify contextual auth headers
- Added workflow to keep `staging` branch in sync with `dev`

# Review By (Date)
- End of Sprint

# Review Tasks

1. Make sure everything in the new netlify-plugin-contextual-auth looks to be on the up and up (I tested it using local branch deploys for this branch: [deployed site](https://6189bf2c8fc9fe0008f5a19b--stanford-alumni.netlify.app/), [deploy log](https://app.netlify.com/sites/stanford-alumni/deploys/6189bf2c8fc9fe0008f5a19b))
2. Check out the new workflow code for merging dev to staging

### CODE REVIEW

File: plugins/netlify-plugin-contextual-auth/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - name: context
+    description: Which context should
+  - name: branch
+    description: Optional branch to match against for context 
+  - name: path
+    description: Header path match string for basic auth
+    default: /*
+  - name: file
+    description: path to `_headers` file
+    default: 'public/_headers'
```
Comment (mattanglin) [inline plugins/netlify-plugin-contextual-auth/manifest.yml:18]: This seemed like something we want be able to make use of elsewhere 🤷🏼 ?

Comment (yvonnetangsu) [reply plugins/netlify-plugin-contextual-auth/manifest.yml:18]: This is nice!

**Review by yvonnetangsu — APPROVED** [review]
Looks great to me! Thanks for setting this up. I'm going to let @sherakama have the final approval since he's much more familiar with this 😃

**Comment by mattanglin** [comment]
Additional Note: Once staging has been deployed I will also setup a "Deploy Staging" Task within Storyblok that interacts with a "build staging" webhook in Netlify so that the client can redeploy the staging site as they update content.

**Pull request merged**